src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,165 @@
+
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM shopping_carts c "
+                    + "LEFT JOIN shopping_carts_products cp ON c.id = cp.cart_id "
+                    + "LEFT JOIN products p on cp.product_id = p.id "
+                    + "WHERE c.user_id = ? AND c.deleted = FALSE");
+            statement.setLong(1, userId);
+            return getShoppingCartFromStatement(statement);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting shopping cart for user with id "
+                                              + userId + " failed", e);
+        }
+    }
+
+    @Override
+    public Optional<ShoppingCart> getById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM shopping_carts c "
+                    + "JOIN shopping_carts_products cp ON c.id = cp.cart_id "
+                    + "JOIN products p on cp.product_id = p.id "
+                    + "WHERE c.id = ? AND c.deleted = FALSE");
+            statement.setLong(1, id);
+            return getShoppingCartFromStatement(statement);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting shopping cart with id "
+                                              + id + " failed", e);
+        }
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM shopping_carts c "
+                    + "JOIN shopping_carts_products cp ON c.id = cp.cart_id "
+                    + "JOIN products p on cp.product_id = p.id "
+                    + "WHERE c.deleted = FALSE",
+                    ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE);
+            ResultSet resultSet = statement.executeQuery();
+            List<ShoppingCart> carts = new ArrayList<>();
+            while (resultSet.next()) {
+                carts.add(getShoppingCartFromResultSet(resultSet));
+                resultSet.previous();
+            }
+            return carts;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get items from DB", e);
+        }
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteProductsFromDB(shoppingCart.getId(), connection);
+            addProductsToDB(shoppingCart, connection);
+            PreparedStatement statement = connection.prepareStatement(
+                    "UPDATE shopping_carts SET user_id = ? "
```

Comment:
can you update the user in the shopping cart if it should be created automatically? I think it makes no sense

Reply:
same as order